### ISSUE TITLE
Should we host a code and learn in Montreal

### ISSUE THREAD

**Issue by mcollina** [nodejs/admin#432]
The team organizing the conference is thinking about cancelling the code&learn given the low engagement from the projects. Do we want to hold it, or can we release that time to the Collab Summit?

@nodejs/tsc @nodejs/community-committee

**Comment by bnb** [comment]
Hard yes if it's possible. Specifically, I know at least in Colombia we basically ran out of things that could be committed by first timers since our test coverage is so well covered at this point. If we can still generate the necessary tasks – or take a different approach, like docs – I'm a hard -1 on skipping it.

**Comment by mcollina** [comment]
The key problem is that there is no one leading it right now. If there is a volunteer to prep the task on our side, we can host it.

cc @Trott @BridgeAR @bengl who helped a lot in past editions.

**Comment by sam-github** [comment]
OK, that makes sense, its a lot of work. I can't volunteer to organize either. I understand the challenge now.

**Comment by gabrielschulhof** [comment]
In N-API there would be a lot of small things to contribute, namely to
write tests that pass NULL to all the parameters accepted by API calls.
Here's an example:

https://github.com/nodejs/node/blob/master/test/js-native-api/test_constructor/test.js#L55-L62

I could probably make up at least 10-15 tasks to hand out on slips of
paper. I don't know what level of difficulty they would qualify as, but the
tasks are definitely not JS-only.

**Comment by mhdawson** [comment]
This issue should probably have more of the background: https://github.com/openjs-foundation/cross-project-council/issues/287.

In that issue it was discussed that we'd pretty much run out of issues to be used for a Node.js code and learn.  The proposal was to make it cover all of the Projects in the OpenJS Foundation:

Some of the proposals are here: https://github.com/openjs-foundation/code-and-learn/issues which includes one for adding tests for N-API in Node.js but that on it's own is not going to be enough.  

If there is not enough interest from the other projects then given that we don't have a volunteer/ability to run a Node.js focussed Code and Learn we may need to skip this time and release the time back to the collab summit.

**Comment by Trott** [comment]
TL;DR I think we should reclaim the time for the Collaborator Summit this time around.

Can't speak for others who might be well-positioned to organize, but I simply don't have the time to do the upfront work required for a successful Code & Learn at this time. Maybe/hopefully next year. I have other reservations about doing a Code & Learn, but if I'm being honest, the biggest one is my own lack of availability for preparation. Doing a *great* Code & Learn with a lot of attendees is a *lot* of upfront work.

**Comment by nschonni** [comment]
Maybe reach out to https://nodeschool.io/montreal/, not sure if they're still actively meeting though

**Comment by Trott** [comment]
Based on email with the Foundation folks/Summit planners., I think the plan is to reclaim the time for Collaborator Summit. I don't think this issue needs to remain open and I don't think it needs to be on the TSC agenda this week. Do you agree, @mcollina? Happy to discuss it if my impression is not everyone else's impression.

**Comment by joyeecheung** [comment]
FWIW we have a lot of left over tasks from the C+L in Shanghai last week since we had a low turnout (it was not a Node or JS conference) and we did not have WiFi at the venue. But this should be pretty useful for future C+L if people run out of tasks (I think these clues can be used to generate ~300 PRs at least)

**Comment by mcollina** [comment]
Do we want to go ahead with the C & L then? Would you like to coordinate/run the session? 

Or should we cancel?

**Comment by Trott** [comment]
Another reason to be cautious about running a Code & Learn this year is that our test reliability in CI is really not up to it. Even when CI is in great shape, a big Code & Learn event causes issues. (If a planned Code & Learn acts as motivation to solve those test reliability issues, then great! But I wouldn't rely on that happening.)

**Comment by joyeecheung** [comment]
> Do we want to go ahead with the C & L then? Would you like to coordinate/run the session?

I will leave early from the conference so I cannot help with the session myself. Just putting the information out there in case anyone is interested...but without someone actually running it I think we should cancel.

**Comment by Trott** [comment]
From today's TSC minutes:

> discussed in today's TSC meeting, no volunteers. Since we
    need to provide an answer unless there is a volunteer to run it by end of day
    today (Nov 6) we'll let the conference know they can get the time back.
    Matteo will let the conference org's know based on issue.

I've removed the `tsc-agenda` label.

**Comment by Fishrock123** [comment]
I personally feel like 'Code & Learn's are fun but fundamentally unproductive.

I'd rather have a much smaller group of more dedicated people who we can do deeper dives with.
Or, like, just host collab summit at that time and hope those folks with that level of interest find their way to showing up. If you're at a collab summit you're probably bound to learn a lot of in-depth stuff, honestly.

**Comment by Trott** [comment]
I closed this even though Community Committee didn't appear to weigh in much. Feel free to reopen if CommComm wants to put it on their agenda or thinks there's more to discuss here or whatever. Just tidying up. Not feeling strongly that this issue must be closed right now or anything.
